forui/lib/src/widgets/line_calendar.dart

```diff
@@ -0,0 +1,352 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+import 'package:meta/meta.dart';
+import 'package:sugar/sugar.dart';
+
+import 'package:forui/forui.dart';
+import 'package:forui/src/foundation/tappable.dart';
+import 'package:forui/src/foundation/util.dart';
+
+const _days = ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun'];
+final _epoch = LocalDate(2000);
+const _textSpacing = 2.0;
+
+/// A calendar that can be scrolled horizontally.
+class FLineCalendar extends StatefulWidget {
+  /// The style. Defaults to [FThemeData.lineCalendarStyle].
+  final FLineCalendarStyle? style;
+
+  /// The currently selected [LocalDate].
+  final ValueNotifier<LocalDate> selected;
+
+  /// The first date in this calendar carousel. Defaults to 1st January 2000.
+  final LocalDate epoch;
+
+  /// Today's date. Defaults to the [LocalDate.now].
+  final LocalDate today;
+
+  /// Creates a [FLineCalendar].
+  FLineCalendar({
+    required this.selected,
+    LocalDate? epoch,
+    LocalDate? today,
+    this.style,
+    super.key,
+  })  : epoch = epoch ?? _epoch,
+        today = today ?? LocalDate.now();
+
+  @override
+  State<FLineCalendar> createState() => _FLineCalendarState();
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties
+      ..add(DiagnosticsProperty('style', style))
+      ..add(DiagnosticsProperty('selected', selected))
+      ..add(DiagnosticsProperty('epoch', epoch))
+      ..add(DiagnosticsProperty('today', today));
+  }
+}
+
+class _FLineCalendarState extends State<FLineCalendar> {
+  late ScrollController _controller = ScrollController();
```

Comment:
I don't think it is possible since I need to set the ScrollController's `initialScrollOffset` to scroll the lineCalendar to the selected date. If not, it will show the start date by default.